beacon-chain/operations/blstoexec/pool.go

```diff
@@ -23,6 +23,8 @@ type PoolManager interface {
 	InsertBLSToExecChange(change *ethpb.SignedBLSToExecutionChange)
 	MarkIncluded(change *ethpb.SignedBLSToExecutionChange) error
 	ValidatorExists(idx types.ValidatorIndex) bool
+	Copy() PoolManager
+	NumPending() int
```

Comment:
If we moved the cycling logic into the pool type we could keep these out of the interface.